processor/src/host/mod.rs

```diff
@@ -131,41 +108,90 @@ where
     }
 }
 
+// HOST LIBRARY
+// ================================================================================================
+
+pub trait HostLibrary {
+    // Returns all event handlers
+    fn get_event_handlers<A>(&self) -> Vec<Box<dyn EventHandler<A>>>
+    where
+        A: AdviceProvider + 'static;
+
+    fn get_mast_forest(&self) -> Arc<MastForest>;
+}
```

Comment:
Should we try to move this to the `miden-core` crate? This way, `stdlib` does not need to depend on `miden-processor`. I think this will require moving `AdviceProvider` trait to `miden-core` as well, but maybe that's OK (we already have some advice provider related structs there).

Also, question: why do we need `+ static` as a trait bound here? Are there any implications to this?

Reply:
> Should we try to move this to the miden-core crate? This way, stdlib does not need to depend on miden-processor.

Agreed, I was indeed planning on doing that. We also don't want user libraries to have to depend on `miden-processor`.

> Also, question: why do we need + static as a trait bound here? Are there any implications to this?

It's required because the `AdviceProvider` is being put on the heap with that `Box<dyn EventHandler<A>>`, so we have to make sure that the concrete implementation is not holding a reference to a short lifetime that could become dangling (enforced by the borrow-checker obviously).

Concretely, this means that we can't have lifetimes in `AdviceProvider` implementations, i.e. no `struct MemAdviceProvider<'a> { ... }`, which is not a problem in practice.